Instagram/ComposeViewController.m

```diff
@@ -0,0 +1,75 @@
+#import "ComposeViewController.h"
+#import "Post.h"
+
+@interface ComposeViewController ()
+
+@property (strong, nonatomic) IBOutlet UIImageView *photo;
+@property (strong, nonatomic) IBOutlet UITextView *caption;
+- (IBAction)didTapCancel:(UIBarButtonItem *)sender;
+- (IBAction)didTapSubmit:(UIBarButtonItem *)sender;
+
+@end
+
+@implementation ComposeViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    self.caption.layer.borderWidth = 2.0f;
+    self.caption.layer.borderColor = [[UIColor lightGrayColor] CGColor];
+    self.caption.layer.cornerRadius = 5;
+
+    UIImagePickerController *imagePickerVC = [UIImagePickerController new];
+    imagePickerVC.delegate = self;
+    imagePickerVC.allowsEditing = YES;
+
+    if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
+        imagePickerVC.sourceType = UIImagePickerControllerSourceTypeCamera;
+    } else {
+        NSLog(@"Camera 🚫 available so we will use photo library instead");
+        imagePickerVC.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+    }
+
+    [self presentViewController:imagePickerVC animated:YES completion:nil];
+
+}
+
+- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary<NSString *,id> *)info {
+
+    // Get the image captured by the UIImagePickerController
+    UIImage *originalImage = info[UIImagePickerControllerOriginalImage];
+    UIImage *editedImage = info[UIImagePickerControllerEditedImage];
+
+    // Do something with the images (based on your use case)
+    [self.photo setImage:originalImage];
+    
+    // Dismiss UIImagePickerController to go back to your original view controller
+    [self dismissViewControllerAnimated:YES completion:nil];
+}
+
+- (IBAction)didTapSubmit:(UIBarButtonItem *)sender {
+    [Post postUserImage:self.photo.image withCaption:self.caption.text withCompletion:^(BOOL succeeded, NSError * _Nullable error) {
+        if (error != nil) {
+            // TODO: handle error
+            NSLog(@"Error posting picture: %@", error.localizedDescription);
```

Comment:
Nit: these are really just for debugging. I'd keep the TODO but lose the logging.